cupy/core/core.pyx

```diff
@@ -2381,35 +2381,46 @@ cpdef ndarray _repeat(ndarray a, repeats, axis=None):
 
     """
     cdef ndarray ret
+
+    # Scalar and size 1 'repeat' arrays broadcast to any shape, for all
+    # other inputs the dimension must match exactly.
+    broadcast = False
     if isinstance(repeats, int):
         if repeats < 0:
             raise ValueError(
                 "'repeats' should not be negative: {}".format(repeats))
-        if axis is None:
-            a = a.reshape((-1, 1))
-            ret = ndarray((a.size, repeats), dtype=a.dtype)
-            if ret.size:
-                ret[...] = a
-            return ret.ravel()
-
-        repeats = [repeats] * a._shape[axis % a._shape.size()]
+        broadcast = True
     elif cpython.PySequence_Check(repeats):
         for rep in repeats:
             if rep < 0:
                 raise ValueError(
                     "all elements of 'repeats' should not be negative: {}"
                     .format(repeats))
-        if axis is None:
-            raise ValueError(
-                "'axis' should be specified if 'repeats' is sequence")
-        if a.shape[axis] != len(repeats):
-            raise ValueError(
-                "'repeats' and 'axis' of 'a' should be same length: {} != {}"
-                .format(a.shape[axis], len(repeats)))
+        if len(repeats) == 1:
+            repeats = repeats[0]
+            broadcast = True
     else:
         raise ValueError(
             "'repeats' should be int or sequence: {}".format(repeats))
 
+    if axis is None:
+        if broadcast:
+            a = a.reshape((-1, 1))
+            ret = ndarray((a.size, repeats), dtype=a.dtype)
+            if ret.size:
+                ret[...] = a
+            return ret.ravel()
+        else:
+            a = a.ravel()
+            axis = 0
+
+    if broadcast:
+        repeats = [repeats] * a._shape[axis % a._shape.size()]
```

Comment:
Do we need to check if `axis` is in the range `-ndim <= axis < ndim`?

Reply:
When `broadcast` is `True`, `repeats` is `int`, right? Can you write a comment?

Reply:
I added axis boundary check.

I fixed to wrap `repeats` as list if it is int, instead of adding comment. I feel it is more intuitive.